src/ProductView/DetailArea/DetailArea.js

```diff
@@ -3,6 +3,25 @@ import ProductDetails from './ProductDetails/ProductDetails.js';
 import StatusDerivation from './StatusDerivation/StatusDerivation.js';
 
 class DetailArea extends Component {
+    constructor(props) {
+        super(props);
+        this.setConfiguration(this.props);
```

Comment:
Yes, it seems to be necessary, I looked it up and the [documentation](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) says:

> React doesn't call componentWillReceiveProps with initial props during mounting. It only calls this method if some of component's props may update. Calling this.setState generally doesn't trigger componentWillReceiveProps.